File: 645-set-mismatch/set-mismatch.py
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        lookup = Counter(nums)
        result = [0, 0]
        N = len(nums)

        for i in range(1, N + 1):
            if (i not in lookup):
                result[1] = i
            elif (lookup[i] == 2):
                result[0] = i

        return result

File: 645-set-mismatch/set-mismatch.js
var findErrorNums = function(nums) {
    let k = 1
    let lookup = new Set(nums)
    let counter = {}
    let n = nums.length
    let missing

    for (let num of nums) {
        if (!(num in counter)) counter[num] = 0
        counter[num] += 1
    }

    for (let i = 1; i <= n; i++) {
        if (!(k in counter)) missing = k
        if (counter[i] == 2) duplicate = i
        k += 1
    }

    return [duplicate, missing]
};